### ISSUE TITLE
ArgumentError: wrong number of arguments (given 3, expected 2) When used in conjunction with redis-store

### ISSUE THREAD

**Issue by sho-work** [sds/mock_redis#292]
Hi!
I'm sorry, I'm new here.
When using the redis-store gem together with mock_redis, I encountered the following error:

```bash
ArgumentError:
       wrong number of arguments (given 3, expected 2)
     # /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis/string_methods.rb:209:in `set'
     # /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis/multi_db_wrapper.rb:22:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis/expire_wrapper.rb:17:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis/transaction_wrapper.rb:30:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis/pipelined_wrapper.rb:27:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis.rb:99:in `block in method_missing'
# /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis.rb:155:in `logging'
     # /usr/local/bundle/gems/mock_redis-0.34.0/lib/mock_redis.rb:98:in `method_missing'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:49:in `block (2 levels) in write_session'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/redis/rack/connection.rb:24:in `with'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:82:in `with'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:49:in `block in write_session'
```

Upon investigating, it seems that the issue stems from the MockRedis::StringMethods module not being able to accept three positional arguments for the set method. Internally, the Rack::Session::Redis class's write_session method is passing a third positional argument, options.to_hash.
Reference: https://github.com/redis-store/redis-rack/blob/master/lib/rack/session/redis.rb#L49
```ruby
def set(key, value, ex: nil, px: nil, exat: nil, pxat: nil, nx: nil, xx: nil, keepttl: nil, get: nil)
```
https://github.com/sds/mock_redis/blob/main/lib/mock_redis/string_methods.rb#L215

Therefore, I propose the following modification:
```ruby
def set(key, value, _hash, ex: nil, px: nil, exat: nil, pxat: nil, nx: nil, xx: nil, keepttl: nil, get: nil)
```

A similar issue was raised in the past, and the conclusion was that it should be resolved on the redis-store side. However, since it has not been resolved on the redis-store, I'm wondering if it might be acceptable to make the modification on the mock_redis side.
https://github.com/sds/mock_redis/issues/266

Would this be feasible, considering the impact and backward compatibility?

**Comment by sds** [comment]
Hey @sho-work, sorry for the delay in response.

The version of `mock_redis` in your stack trace suggests you are on an older version. Can you confirm this problem still happens with the latest version? ([0.42.0](https://github.com/sds/mock_redis/releases/tag/v0.42.0) at time of writing)

Thanks!

**Comment by sho-work** [comment]
Hey @sds, Thank you for your response!

I tried upgrading `mock_redis` to version 0.42.0, but I'm still encountering an ArgumentError similar to the one below.
```bash
ArgumentError:
       wrong number of arguments (given 3, expected 2)
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis/string_methods.rb:215:in `set'
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis/multi_db_wrapper.rb:21:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis/expire_wrapper.rb:17:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis/transaction_wrapper.rb:30:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis/pipelined_wrapper.rb:27:in `method_missing'
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis.rb:94:in `block in method_missing'
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis.rb:150:in `logging'
     # /usr/local/bundle/gems/mock_redis-0.42.0/lib/mock_redis.rb:93:in `method_missing'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:49:in `block (2 levels) in write_session'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/redis/rack/connection.rb:24:in `with'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:82:in `with'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:49:in `block in write_session'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:70:in `with_lock'
     # /usr/local/bundle/gems/redis-rack-2.1.4/lib/rack/session/redis.rb:48:in `write_session'
```

**Comment by sds** [comment]
Thanks for confirming. I am supportive of making the change, though I would think we would need to have `_hash = nil` in the signature so that any callers who don't supply the magical 3rd argument don't fail either.

If you open a PR with passing tests, happy to merge and cut a release. Thanks!

**Comment by sho-work** [comment]
@sds 
Thank you! I have made the modifications in the following pull request. I would appreciate it if you could review it! 
- https://github.com/sds/mock_redis/pull/299

**Comment by sds** [comment]
Released in [0.43.0](https://github.com/sds/mock_redis/releases/tag/v0.43.0). Thanks again.
